04. 완전탐색(블루투포스)/01. 자리수의 합/정봉찬/index.js

```diff
@@ -1,5 +1,39 @@
+/*
+✅문제 제목: 자리수의 합
+
+✅문제 유형: 완전 탐색(블루투포스)
+
+✅문제 풀이 날짜: 2024-07-19
+
+💡문제 분석 요약
+  - 자연수의 자릿수의 합을 구한다. 그 합이 최대인 자연수를 출력한다.
+  - 자릿수의 합이 같은 경우 원래 숫자가 큰 숫자를 출력한다.
+
+💡알고리즘 설계
+  - 자연수 배열을 반복문 돌면서 자릿수의 합이 최대일 때 자연수의 크기를 비교한다.
+*/
+
 function solution(n, arr) {
   let answer;
+  let maxKey = 0;
+  let maxValue = 0;
+
+  for (let i = 0; i < n; i++) {
+    const current = arr[i];
+    const sum = +[...current.toString()].reduce((pre, cur) => pre + +cur, 0);
```

Comment:
👍🏻 reduce 메서드 초기값 명시해준 것 좋네요 좋은 풀이 퍼가요

Reply:
추가로 reduce가 숫자를 반환해서 앞에 + 없어도 될 거 같아요
```suggestion
    const sum = [...current.toString()].reduce((pre, cur) => pre + +cur, 0);
```